warehouse/core/src/main/java/datawave/util/compression/OptionCompressor.java

```diff
@@ -0,0 +1,236 @@
+ * <p>
+ * General process:
+ * </p>
+ * <ol>
+ * <li><b>Compression</b>: Original Data → Compress with Algorithm (selected via {@link CompressionMethod}) → Encode in Base64 (except {@code NONE}) → Return as
+ * String</li>
+ * <li><b>Decompression</b>: Input String (Base64 if compressed; plain text if {@code NONE}) → Decode Base64 (if applicable) → Decompress with Algorithm →
+ * Return Original Data</li>
+ * </ol>
+ *
+ * <p>
+ * Base64 encoding ensures that compressed data can be safely stored or transferred as text. For {@link CompressionMethod#NONE} the data is returned as-is and
+ * is not Base64-encoded.
+ * </p>
+ */
+public class OptionCompressor {
+
+    // possible data class for holding metadata for the compression method. Not sure if I'll need to actually include this.
+    // see Gz.java or Bzip2.java from Accumulo.
+    // public CompressionAlgorithmConfiguration configuration;
+
+    // additionally, see https://commons.apache.org/proper/commons-compress/
+    // it's probably best to source all the compression from the same place if possible.
+
+    public enum CompressionMethod {
+        NONE, GZIP, BZIP2, SEVEN_ZIP
+    }
+
+    /**
+     * Compresses the given string using the specified {@link CompressionMethod}.
+     *
+     * <p>
+     * Order of operations (by method):
+     * </p>
+     * <ul>
+     * <li>{@code NONE}: Return {@code data} unchanged; no Base64 applied.</li>
+     * <li>{@code GZIP}: Convert {@code data} to bytes with {@code charset} → GZIP-compress → Base64-encode → return String.</li>
+     * <li>{@code BZIP2}: Convert {@code data} to bytes with {@code charset} → BZIP2-compress → Base64-encode → return String.</li>
+     * <li>{@code SEVEN_ZIP}: Convert {@code data} to bytes with {@code charset} → LZMA-compress (7z algorithm stream) → Base64-encode → return String.</li>
+     * </ul>
+     */
+    public String compress(final String data, final CompressionMethod method, final Charset charset) throws IOException {
+        switch (method) {
+            case NONE:
+                return data;
+            case GZIP:
+                return compressGZIP(data, charset);
+            case BZIP2:
+                return compressBZIP2(data, charset);
+            case SEVEN_ZIP:
+                return compress7ZIP(data, charset);
+            default:
+                throw new IllegalArgumentException("unrecognized compression option: " + method);
+        }
+    }
+
+    /**
+     * GZIP implementation reference: 1) String → bytes via charset 2) Write to GZIPOutputStream (must close to finalize) 3) Base64-encode compressed bytes
+     */
+    private String compressGZIP(final String data, final Charset charset) {
+        final byte[] input = data.getBytes(charset);
+
+        try (ByteArrayOutputStream baos = new ByteArrayOutputStream(); GZIPOutputStream gzip = new GZIPOutputStream(baos)) {
+
+            gzip.write(input);
+            gzip.close(); // must close to flush all compressed data into baos
```

Comment:
If you need the gzip stream to e closed you can always nest another try-with-resources block. Same comment goes for other instances in this class